packages/client/src/authentication/__helpers__/setupAuthentication.ts

```diff
@@ -1,41 +1,203 @@
+/* eslint-disable no-console */
 import { Wallet } from 'ethers';
 
 import { mumbaiSandbox } from '../../consts/environments';
+import { Profile } from '../../profile';
+import { getTxIdFromRelayerResult, Transaction } from '../../transaction';
 import { Authentication } from '../Authentication';
-import { setupTestWallet } from './setupTestWallet';
+import { signAndBroadcast } from './signAndBroadcast';
 
 const testConfig = {
   environment: mumbaiSandbox,
 };
 
-export function setupAuthentication() {
-  let auth: Authentication;
+type TestSetup = {
+  authentication: Authentication;
+  profileId: string;
+  wallet: Wallet;
+  walletAddress: string;
+};
+
+type SetupOptions = {
+  withNewProfile?: boolean;
+  withDispatcher?: boolean;
+};
+
+const defaultOptions: SetupOptions = {
+  withNewProfile: false,
+  withDispatcher: false,
+};
+
+const DOES_BURN_PROFILE_WORK = false;
+
+export function setupAuthentication(options?: SetupOptions): () => TestSetup {
```

Comment:
As it just does not return the authentication module but all relevant data to run authenticated scenarios I would rename this into:

```suggestion
export function setupAuthenticatedScenario(options?: SetupOptions): () => TestSetup {
```